testng-core/src/main/java/org/testng/internal/BaseTestMethod.java

```diff
@@ -176,6 +178,34 @@ public String[] getMethodsDependedUpon() {
     return m_methodsDependedUpon;
   }
 
+  @Override
+  public Set<ITestNGMethod> downstreamDependencies() {
+    return Collections.unmodifiableSet(downstreamDependencies);
+  }
+
+  @Override
+  public Set<ITestNGMethod> upstreamDependencies() {
+    return Collections.unmodifiableSet(upstreamDependencies);
+  }
+
+  public void setDownstreamDependencies(Set<ITestNGMethod> methods) {
+    if (RuntimeBehavior.isMemoryFriendlyMode()) {
+      downstreamDependencies =
+          methods.stream().map(LiteWeightTestNGMethod::new).collect(Collectors.toSet());
+    } else {
+      downstreamDependencies.addAll(methods);
```

Comment:
If the method is called more than once, the behavior won't be the same in all cases: sometimes the first setup is lost and sometimes the setup is added.

Just clear before add or create a new set.

Same remark for upstream dep.

Reply:
This is not going to happen. The upstream/downstream dependencies are computed and injected into a test method even before the execution happens. So this setting is called ONLY once per `<test>` tag. But I have still added the edit check to clear the set if its not empty before setting upstream/downstream dependencies.
Please check now